android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/LanguageResource.java

```diff
@@ -45,5 +64,71 @@ public int hashCode() {
 
   public abstract Bundle buildDownloadBundle();
 
-  //public abstract boolean equals();
+  /**
+   * Constructor using JSON Objects from installed language resource list
+   * @param installedObj
+   */
+  public LanguageResource(JSONObject installedObj) {
+    this.fromJSON(installedObj);
+  }
+
+  public LanguageResource(JSONObject languageJSON, JSONObject keyboardJSON) {
+    // Noop - needed for Keyboard()
+  }
+
+  public LanguageResource(JSONObject lexicalModelJSON, boolean fromCloud) {
+    // Noop - needed for LexicalModel()
+  }
+
+  /**
+   * Constructor using properties
+   * For now, helpLink not included because it's only used by Keyboard
```

Comment:
I'm a little worried about this -- it gives the parameter-based constructor an uneven mapping with the JSON-based constructor. Is there a good reason to have it in one place and not the other?

Reply:
k, I've moved helpLink back to the base constructor